backend/src/migrations/20230827114312_add_user_details_table.ts

```diff
@@ -0,0 +1,118 @@
+import { type Knex } from 'knex';
+
+const TableName = {
+    USERS: 'users',
+    USER_DETAILS: 'user_details',
+    FILES: 'files',
+};
+const TABLE_NAME = 'user_details';
+
+const ColumnName = {
+    ID: 'id',
+    USER_ID: 'user_id',
+    ROLE: 'role',
+    IS_APPROVED: 'is_approved',
+    DENIED_REASON: 'denied_reason',
+    IS_HIRED: 'is_hired',
+    PROFILE_NAME: 'profile_name',
+    SALARY_EXPECTATION: 'salary_expectation',
+    HIRED_SALARY: 'hired_salary',
+    JOB_TITLE: 'job_title',
+    LOCATION: 'location',
+    EXPERIENCE_YEARS: 'experience_years',
+    EMPLOYMENT_TYPE: 'employment_type',
+    DESCRIPTION: 'description',
+    ENGLISH_LEVEL: 'english_level',
+    NOT_CONSIDERED: 'not_considered',
+    PREFERRED_LANGUAGES: 'preferred_languages',
+    PROJECT_LINKS: 'project_links',
+    PHOTO_ID: 'photo_id',
+    FULL_NAME: 'full_name',
+    PHONE: 'phone',
+    LINKEDIN_LINK: 'linkedin_link',
+    COMPANY_NAME: 'company_name',
+    COMPANY_LOGO_ID: 'photo_id',
+    COMPANY_WEBSITE: 'company_website',
+    EMPOYER_POSITION: 'employer_position',
+    CV_ID: 'cv_id',
+    CREATED_AT: 'created_at',
+    UPDATED_AT: 'updated_at',
+};
+
+const RelationRule = {
+    CASCADE: 'CASCADE',
+    SET_NULL: 'SET NULL',
+} as const;
+
+function up(knex: Knex): Promise<void> {
+    return knex.schema.createTable(TABLE_NAME, (table) => {
+        table.increments(ColumnName.ID).primary();
+        table
+            .integer(ColumnName.USER_ID)
+            .unique()
+            .notNullable()
+            .references(ColumnName.ID)
+            .inTable(TableName.USERS)
+            .onUpdate(RelationRule.CASCADE)
+            .onDelete(RelationRule.CASCADE);
+        table
+            .integer(ColumnName.PHOTO_ID)
+            .references(ColumnName.ID)
+            .inTable(TableName.FILES)
+            .onUpdate(RelationRule.CASCADE)
+            .onDelete(RelationRule.CASCADE);
+        table.boolean(ColumnName.IS_APPROVED).defaultTo(false);
+        table.string(ColumnName.DENIED_REASON);
+        table.boolean(ColumnName.IS_HIRED).defaultTo(false);
+        table.string(ColumnName.PROFILE_NAME);
```

Comment:
In figma this field is required, so please make all fields, which required in figma required in you migrations

Reply:
But in user_details we save data for talent, employer and admin and not all fields required for all of them

Reply:
Oh, yes you can keep it